src/profiling/types.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Resource } from '@opentelemetry/resources';
+
+export interface ProfilingStartOptions {
+  samplingIntervalMicroseconds: number;
+  recordDebugInfo: boolean;
+}
+
+export interface ProfilingStackTrace {
+  /** Timestamp of the sample (nanoseconds since Unix epoch). */
+  timestamp: string;
+  /** Formatted stacktrace. */
+  stacktrace: string;
+  traceId: string;
+  spanId: string;
+}
+
+export interface ProfilingData {
+  /** Timestamp when profiling was started (nanoseconds since Unix epoch). */
+  startTimeNanos: string;
+  stacktraces: ProfilingStackTrace[];
```

Comment:
Casing difference between `stacktraces` and `ProfilingStackTrace` either consider `stackTraces` or `ProfilingStacktraces` or just call them "stacks" like ECMAscript does.

Reply:
Ah thanks, I initially typed everything as `stackTrace`, but then changed to `stacktrace` and missed the type casings.